### PR TITLE
Profile cache time for performance purpose

### PR DESCRIPTION
**Pull request by vitalyzhakov**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #12775

### CODE REVIEW

**Comment by vitalyzhakov** [comment]
```* @param mixed $key a key identifying the cached value. This can be a simple string or
     * a complex data structure consisting of factors representing the key.```
@samdark , this function really using mixed argument or strings only?

**Comment by samdark** [comment]
Yes, mixed is correct. See `buildKey()`.

File: framework/caching/Cache.php [hunk 1]
```diff
@@ -101,8 +101,11 @@ public function buildKey($key)
      */
     public function get($key)
     {
-        $key = $this->buildKey($key);
-        $value = $this->getValue($key);
+        \Yii::beginProfile($key);
```
Comment (samdark) [inline framework/caching/Cache.php]: `__METHOD__` should be used as a second argument.

Comment (samdark) [inline framework/caching/Cache.php]: Need to check how these calls affect production performance. If much, add a check for `YII_DEBUG`.

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by vitalyzhakov** [comment]
I have changed code and create new PR #12911

**Pull request closed without merging**